runtime/doc/lsp.txt

```diff
@@ -0,0 +1,306 @@
+>
+                        {
+                            'newName': 'exampleRename'
+                        }
+<
+                    Sent:
+>
+                        {
+                            'textDocument': '/home/user/test/rename.lua',
+                            'position': { 'line': 16, 'character': 8 }
+                            'newName': 'exampleRename'
+                        }
+<
+
+            [callback]: An optional callback that will be executed
+            once the response has been received. If this is passed and not
+            |v:null|, then the |lsp-built-in-callback| will not be executed for
+            this request. If you do not set |built-in-callbacks| or this request does not have a
+            a specify filetype callback, then no callback will be executed upon receiving the request.
+
+            The callback must be a string with the name of the vimscript
+            function to call. The signature must be:
+
+                `func_name({success}, {data})`
+
+                where {success} is a boolean where true means successful request
+                and false means the request failed.
+
+                The parameter {data} is a dictionary containing the possible
+                response data (or error) data as specified by the language
+                server protocol.
+
+                The callback will be executed from lua running:
+
+                    `vim.api.nvim_call_function('func_name', { success, data })`
+
+            [filetype]: The filetype associated with the server.
+            This will default to the current |filetype| if nothing is passed.
+
+            Configuration for the default server can be found at
+            |lsp-configuration|.
+
+                                                          *lsp#request_async()*
+lsp#request_async({method} [, {arguments} [, {callback} [, {filetype}]]])
+    Creates and sends a request to the server for [filetype]. This request
+    will not block and will call desired the [callback] upon completion.
+
+    Returns the request ID.
+
+    See |lsp#request()| for more information.
+
+    Example:
+
+autocmd CursorHold *
+    \ :lua require('vim.lsp').plugin.request_async(
+    \     'textDocument/hover',
+    \     require('vim.lsp').util.get_text_document_params(),
+    \     require('vim.lsp').util.get_filetype()
+    \ )
+
+================================================================================
+                                                         *lsp-built-in-callback*
+
+The following requests have built-in callbacks defined to handle the response
+in an idiomatic vim fashion.
+
+  textDocument/definition~
+  textDocument/hover~
+  textDocument/publishDiagnostics~
+  textDocument/references~
+
+require('vim.lsp').config.callbacks.set_all_builtin_callbacks()
+  You can set all built-in callbaks
+
+require('vim.lsp').config..callbacks.set_builtin_callback({method})
+    You can set a specify built-in callback.
```

Comment:
```suggestion
    You can enable a specific built-in callback.
```

First I understood that this would set/override a builtin/default callback.

Reply:
It will surely be read like that. :)